tests/src/ApiClientTest.php

```diff
@@ -4,104 +4,157 @@
-use OpenEuropa\CdtClient\Endpoint\MainEndpoint;
-use OpenEuropa\CdtClient\Endpoint\RequestsEndpoint;
-use OpenEuropa\CdtClient\Endpoint\StatusEndpoint;
-use OpenEuropa\CdtClient\Endpoint\TokenEndpoint;
-use OpenEuropa\CdtClient\Endpoint\ValidateEndpoint;
-use OpenEuropa\CdtClient\Http\Download;
+use GuzzleHttp\Psr7\Response;
+use OpenEuropa\CdtClient\Contract\ApiFactoryInterface;
+use OpenEuropa\CdtClient\Exception\ValidationErrorsException;
+use OpenEuropa\CdtClient\Model\Response\ReferenceData;
 use OpenEuropa\CdtClient\Model\Response\Token;
-use OpenEuropa\Tests\CdtClient\Traits\ClientTestTrait;
+use OpenEuropa\CdtClient\Model\Response\Translation;
+use OpenEuropa\Tests\CdtClient\Traits\ApiTestTrait;
+use OpenEuropa\Tests\CdtClient\Traits\RequestModelTestTrait;
 use PHPUnit\Framework\TestCase;
 
 /**
  * @coversDefaultClass \OpenEuropa\CdtClient\ApiClient
  */
 class ApiClientTest extends TestCase
 {
-    use ClientTestTrait;
+    use ApiTestTrait;
+    use RequestModelTestTrait;
 
-    protected ApiClientInterface $client;
-
-    protected function setUp(): void
+    /**
+     * @covers \OpenEuropa\CdtClient\ApiClient
+     * @covers \OpenEuropa\CdtClient\ApiFactory::setToken
+     */
+    public function testTokenSetter(): void
     {
-        $this->client = $this->getTestingClient();
+        $client = $this->getTestingApiClient();
+
+        $token = new Token();
+        $token->setAccessToken('testtoken');
+        $client->setToken($token);
+
+        // Use reflection to access the protected property.
+        $apiClientReflection = new \ReflectionClass($client);
+        $apiFactoryProperty = $apiClientReflection->getProperty('apiFactory');
+        $apiFactory = $apiFactoryProperty->getValue($client);
+        assert($apiFactory instanceof ApiFactoryInterface);
+
+        $apiFactoryReflection = new \ReflectionClass($apiFactory);
+        $tokenProperty = $apiFactoryReflection->getProperty('token');
+        $actualToken = $tokenProperty->getValue($apiFactory);
+        self::assertEquals($token, $actualToken);
     }
 
     /**
-     * @covers ::createContainer
-     * @covers ::getConfigValue
-     * @covers ::extractConfigValues
+     * @covers ::requestToken
      */
-    public function testContainer(): void
+    public function testRequestToken(): void
     {
-        $container = $this->getClientContainer($this->client);
-
-        // Check container services.
-        $this->assertInstanceOf(TokenEndpoint::class, $container->get('auth'));
-
-        $this->assertInstanceOf(MainEndpoint::class, $container->get('main'));
-        $this->assertInstanceOf(TokenAwareInterface::class, $container->get('main'));
+        $responses = [
+            new Response(200, [], (string) file_get_contents(__DIR__ . '/../fixtures/json/simple_token_call_response.json')),
+        ];
+        $client = $this->getTestingApiClient([], $responses, false);
+        self::assertInstanceOf(Token::class, $client->requestToken());
```

Comment:
The assertions do not care whether the expected results are identical, this type of checking is covered by endpoint testing.